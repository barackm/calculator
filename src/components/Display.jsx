import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div className="display-container">
    <h3 title="result">{result}</h3>
  </div>
);

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};
export default Display;
