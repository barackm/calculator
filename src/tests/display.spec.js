import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Display Component', () => {
  it('Should match spanshot', () => {
    const tree = renderer.create(<Display result="10+10" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
