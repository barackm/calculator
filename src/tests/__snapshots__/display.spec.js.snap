// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component Should match spanshot 1`] = `
<div
  className="display-container"
>
  <h3
    title="result"
  >
    10+10
  </h3>
</div>
`;
